index.js

```diff
@@ -271,34 +269,104 @@ if (require.main === module) {
 
   var html = fs.readFileSync(path.join(__dirname, 'tree-viz.html')).toString();
 
+  var title = Buffer.isBuffer(code) ? 'Buffer' : code;
   html = html.replace('INSERT TREE HERE', JSON.stringify(counts, null, '  '))
-    .replace('INSERT TITLE HERE', args['<script.js>'])
+    .replace('INSERT TITLE HERE', title)
     .replace('INSERT underscore.js HERE', 'data:application/javascript;base64,' + assets.underscoreJs)
     .replace('INSERT webtreemap.js HERE', 'data:application/javascript;base64,' + assets.webtreemapJs)
     .replace('INSERT webtreemap.css HERE', 'data:text/css;base64,' + assets.webtreemapCss);
 
+  return {
+    totalBytes: sizes.totalBytes,
+    numUnmapped: numUnmapped,
+    counts: counts,
+    html: html,
+  };
+}
+
+if (require.main === module) {
+  var args = docopt(doc, {version: '1.5.0'});
+  expandGlob(args);
+  validateArgs(args);
+
+  var html = true;
+
+  if(args['--json'] || args['--tsv']) {
+    html = false;
+  }
+
+  var replace = {};
+  var argsReplace = args['--replace'];
+  var argsWith = args['--with'];
+  if(argsReplace && argsWith) {
+    for(var replaceIndex = 0; replaceIndex < argsReplace.length; replaceIndex += 1) {
+      replace[argsReplace[replaceIndex]] = argsWith[replaceIndex];
+    }
+  }
+
+  try {
+    var data = explore(
+      args['<script.js>'],
+      args['<script.js.map>'],
+      {
+        onlyMapped: args['--only-mapped'] || args['-m'],
+        html: html,
+        noRoot: args['--noroot'],
+        replace: replace,
+      }
+    );
+  } catch(err) {
+    if(err.code === 'ENOENT') {
+      console.error('File not found! -- ', err.message);
+      process.exit(1);
+    } else {
+      console.error(err.message);
+      process.exit(1);
+    }
+  }
+
+  var numUnmapped = data.counts[UNMAPPED];
+  if (numUnmapped) {
+    var totalBytes = data.totalBytes;
+    var pct = 100 * numUnmapped / totalBytes;
+    console.warn(
+      'Unable to map', numUnmapped, '/', totalBytes,
+      'bytes (' + pct.toFixed(2) + '%)');
+  }
+
+
+  if(args['--json']) {
+    console.log(JSON.stringify(data.counts, null, '  '));
+    process.exit(0);
+  }
+
+  if(args['--tsv']) {
+    console.log('Source\tSize');
+    _.each(data.counts, function(source, size) { console.log(size + '\t' + source); });
+    process.exit(0);
+  }
+
   if (args['--html']) {
-    console.log(html);
+    console.log(data.html);
     process.exit(0);
   }
 
   var tempName = temp.path({suffix: '.html'});
-  fs.writeFileSync(tempName, html);
+  fs.writeFileSync(tempName, data.html);
   open(tempName, function(error) {
     if (!error) return;
     console.error('Unable to open web browser.');
     console.error('Either run with --html, --json or --tsv, or view HTML for the visualization at:');
     console.error(tempName);
   });
-
 }
 
+module.exports = explore;
```

Comment:
Was thinking about how nice it might be if the _generation of the html_ was also an exported 
 function. 

```js
import { parse, generateHtml } from 'source-map-explorer';
const { files } = parse(filename);
// third-party function to get versions from the package.json files and append to the filenames
const filesWithVersions = addVersionsToNodeModuleFilenames(files);
const html = generateHtml(files);
```

or

```js
import { parse, generateHtml } from 'source-map-explorer';
const { files } = parse(filename);
// third-party function to find differences in source maps
const fileDiff = getDiffFromPreviousBuild(files);
const html = generateHtml(fileDiff);
```

Thoughts?

Reply:
Hmm, I separated end exported `generateHtml` but I'm not sure if we should make it official and document it.

Reply:
Works for me.